bulk_site_creator/views.py

```diff
@@ -0,0 +1,181 @@
+from canvas_course_site_wizard.models import CanvasSchoolTemplate
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.http import HttpRequest, JsonResponse
+from django.shortcuts import render
+from django.views.decorators.http import require_http_methods
+from django_auth_lti import const
+from django_auth_lti.decorators import lti_role_required
+from icommons_common.canvas_api.helpers import accounts as canvas_api_accounts
+from icommons_common.canvas_utils import \
+    SessionInactivityExpirationRC  # TODO replace this
+from icommons_common.models import (  # TODO: update to coursemanager.models
+    CourseGroup, Department, Term)
+from lti_permissions.decorators import lti_permission_required
+
+from common.utils import (get_canvas_site_template,
+                          get_canvas_site_templates_for_school,
+                          get_term_data_for_school)
+
+from .schema import JobRecord
+from .utils import (batch_write_item, generate_task_objects,
+                    get_course_instance_query_set,
+                    get_course_instances_without_canvas_sites,
+                    get_department_name_by_id, get_term_name_by_id)
+
+logger = logging.getLogger(__name__)
+
+SDK_CONTEXT = SessionInactivityExpirationRC(**settings.CANVAS_SDK_SETTINGS)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["GET", "POST"])
+def index(request):
+    sis_account_id = request.LTI["custom_canvas_account_sis_id"]
+    term_ids, _current_term_id_id = get_term_data_for_school(sis_account_id)
+    school_id = sis_account_id.split(":")[1]
+    canvas_site_templates = get_canvas_site_templates_for_school(school_id)
+    potential_course_sites_query = None
+
+    if request.method == "POST":
+        selected_term_id = request.POST["course-term_id"]
+        # Retrieve all course instances for the given term_id and account that do not have Canvas course sites
+        # nor are set to be fed into Canvas via the automated feed
+        potential_course_sites_query = get_course_instance_query_set(
+            selected_term_id, sis_account_id
+        ).filter(canvas_course_id__isnull=True, sync_to_canvas=0)
+
+    # TODO maybe better to use template tag unless used elsewhere?
+    potential_course_site_count = (
+        potential_course_sites_query.count() if potential_course_sites_query else 0
+    )
+
+    context = {
+        "term_ids": term_ids,
+        "potential_course_sites": potential_course_sites_query,
+        "potential_site_count": potential_course_site_count,
+        "canvas_site_templates": canvas_site_templates,
+    }
+    return render(request, "bulk_site_creator/index.html", context=context)
+
+
+@login_required
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["POST"])
+def create_bulk_job(request: HttpRequest) -> Optional[JsonResponse]:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 61 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 122-180 into a separate function.